### PR TITLE
FEATURE - handle changeset scenarios for 412 warnings

### PR DESCRIPTION
**Pull request by otsquest101**
- changes made to group  multiple 412 responses for  batch request in changeset mode  with a single 412 error response
- no changes required for isolated mode
- in case of one 200 ok and other multiple 412 responses in a changeset batch request, show both 200 and 412 as a single error response.

### CODE REVIEW

File: packages/fe-mockserver-core/src/middleware.ts [hunk 1]
```diff
@@ -126,3 +126,26 @@ export async function createMockMiddleware(
...
     await Promise.all(oDataHandlerPromises);
 }
+
+/**
+ * Exracts JSON response from multipart content in case of changeset batch
+ *
+ * @param batchResponse
+ */
+
+export function getJsonFromMultipartContent(batchResponse: string) {
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/middleware.ts:136]: This seems to be only test code, can you please ensure it's only in test sources ?

File: packages/fe-mockserver-core/src/router/batchRouter.ts [hunk 2]
```diff
@@ -7,26 +7,104 @@ import type { Batch, BatchPart } from './batchParser';
...
 export function isPartChangeSet(part: BatchPart | Batch): part is Batch {
     return (part as Batch).isChangeSet;
 }
+
+/**
+ * Returns whether the response returns a 412 or not.
+ * @param partRequest
+ * @returns {boolean}
+ */
+function isResponse412(partRequest: ODataRequest) {
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/router/batchRouter.ts:39]: this is a one liner that is called once, please donc create a function for it

File: packages/fe-mockserver-core/src/router/batchRouter.ts [hunk 3]
```diff
@@ -63,21 +190,29 @@ export function batchRouter(dataAccess: DataAccess): NextHandleFunction {
...
                             changeSetPart as BatchPart,
                             dataAccess,
                             part.boundary,
                             req.tenantId!,
-                            globalHeaders
+                            globalHeaders,
+                            true
                         );
+                        if (batchPartRes !== undefined) {
+                            // may be make it null in case of 412
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/router/batchRouter.ts:210]: Maybe or maybe not ?

**Review by nlunets — CHANGES_REQUESTED** [review]
Can you please describe in the pull request the different scenarios that should be supported and how they are supposed to work ? 

The whole concept of grouping the 412 response is pretty unclear to me and the code might need further tweaking based on the details.

- Multiple request in changeset or isolated mode
- Some part failing with 412, other ok, other failing with 500.

File: packages/fe-mockserver-core/src/middleware.ts (hunk 1 above)
Comment (otsquest101) [reply packages/fe-mockserver-core/src/middleware.ts]: done

File: packages/fe-mockserver-core/src/router/batchRouter.ts (hunk 2 above)
Comment (otsquest101) [reply packages/fe-mockserver-core/src/router/batchRouter.ts]: done

File: packages/fe-mockserver-core/src/router/batchRouter.ts (hunk 3 above)
Comment (otsquest101) [reply packages/fe-mockserver-core/src/router/batchRouter.ts]: removed comment and changed to null comparison

**Comment by otsquest101** [comment]
> Can you please describe in the pull request the different scenarios that should be supported and how they are supposed to work ?
> 
> The whole concept of grouping the 412 response is pretty unclear to me and the code might need further tweaking based on the details.
> 
> * Multiple request in changeset or isolated mode
> * Some part failing with 412, other ok, other failing with 500.

Hi @nlunets I have resolved the comments, added description for the  412 scenarios addressed with this PR. There is one scenario of multiple errors (multiple 412s and 500.) for which I am following up with models and backend colleagues. If you agree, this can be added as a follow up PR to this one.

**Review by nlunets — APPROVED** [review]

**Pull request merged**